lib/sdf/conversions.rb

```diff
@@ -46,16 +46,16 @@ def self.vector3_to_eigen(vector3)
 
         # Converts a SDF boolean into a Ruby true/false value
         #
-        # @param [String,#text] pose the pose as "x y z" (as expected in SDF) or
-        #   nil, in which case a zero translation is returned
+        # @param [String,#text] boolean the SDF boolean ('true','false','0',1')
         # @return [Boolean]
         def self.to_boolean(xml)
             if xml.respond_to?(:text)
                 xml = xml.text
             end
-            if xml == 'true'
+            xml.gsub!(" ","")
```

Comment:
This is much better written as

```
xml = xml.strip
```

which would also not convert "fa ls     e" to "false"